### DISCUSSION TITLE
Different results using DefaultPredictor vs model(inputs)

### DISCUSSION THREAD

**Discussion by nrasiwas** [facebookresearch/detectron2#3097] [Q&A]
Getting two different results on the same image using DefaultPredictor vs model(inputs). Here are the two different outputs

![image](https://user-images.githubusercontent.com/8678949/119783023-71743d00-beea-11eb-9f6f-5ca71b3d49d5.png)
![image](https://user-images.githubusercontent.com/8678949/119783089-80f38600-beea-11eb-9ed8-b512b0661dd6.png)

Code snippets:
```
im = cv2.imread("...")
outputs = predictor(im)
print(outputs["instances"].pred_classes)
print(outputs["instances"].pred_boxes)
```

Using model:
```
from detectron2.modeling import build_model
model = build_model(cfg)  # returns a torch.nn.Module

from detectron2.checkpoint import DetectionCheckpointer
DetectionCheckpointer(model).load(cfg.MODEL.WEIGHTS)  # load a file, usually from cfg.MODEL.WEIGHTS

image = torch.as_tensor(im.transpose(2, 0, 1))
inputs = [{'image': image}]

model.eval()
with torch.no_grad():
    outputs = model(inputs)

print(outputs[0]["instances"].pred_classes)
print(outputs[0]["instances"].pred_boxes)
```

**Comment by ljtruong** [comment]
The difference is preprocessing. 

This is a reference to the [defaultpredictor](https://github.com/facebookresearch/detectron2/blob/a25edd3e8e06768a5f3fac7c34147a2f90b615f4/detectron2/engine/defaults.py#L310-L317) which shows the preprocessing step before applying model(inputs).

**Reply by nrasiwas** [reply]
Thanks. I was able to replicate the results, although I have a confusion now. My original image was 500x458 and the transform upsamples it to 873x800. Is this really helpful? Processing on the larger image is 3x slower (because it has 3x more pixels), but there is no extra information. 

ResizeTransform(h=500, w=458, new_h=873, new_w=800, interp=2)

Do you think I will loose much by directly operating on 500x458 size images?

**Reply by ljtruong** [reply]
you can change the default input sizes using cfg.INPUT.MIN_SIZE_TRAIN which defaults to 800 as the smallest size. 
The aim of resizing in this manner is about maintaining aspect ratio. 

in your case training on smaller image sizes wouldn't lose anything as the images are smaller than the default resize.

**Comment by nrasiwas** [comment]
I am actually using pertained models from the model zoo. Do you think if
they were trained on 800 as minimum size but I have 500px images, is better
to upscale or its fine to just feed in 500px itself?
